Lib/asyncio/queues.py

```diff
@@ -150,7 +175,12 @@ async def get(self):
         """Remove and return an item from the queue.
 
         If queue is empty, wait until an item is available.
+
+        Raises QueueShutDown if the queue has been shut down and is empty, or
+        if the queue has been shut down immediately.
         """
+        if self._is_shutdown and self.empty():
+            raise QueueShutDown
         while self.empty():
```

Comment:
```suggestion
        while self.empty():
            if self._is_shutdown and self.empty():
                raise QueueShutDown
```

I think I can reduce duplication with this and the next suggestion

Reply:
Ditto.